src/interfaces/IRateLimits.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.8.0;
+
+interface IRateLimits {
+
+    /**********************************************************************************************/
+    /*** Structs                                                                                ***/
+    /**********************************************************************************************/
+
+    /**
+     * @dev Struct representing a rate limit.
+     *      The current rate limit is calculated using the formula:
+     *      `currentRateLimit = min(slope * (block.timestamp - lastUpdated) + lastAmount, maxAmount)`.
+     * @param maxAmount Maximum allowed amount at any time.
+     * @param slope The slope of the rate limit, used to calculate the new limit based on time passed. [tokens / second]
+     * @param lastAmount The amount left available at the last update.
+     * @param lastUpdated The timestamp when the rate limit was last updated.
+     */
+    struct RateLimitData {
+        uint256 maxAmount;
+        uint256 slope;
+        uint256 lastAmount;
+        uint256 lastUpdated;
+    }
+
+    /**********************************************************************************************/
+    /*** Events                                                                                 ***/
+    /**********************************************************************************************/
+
+    /**
+     * @dev Emitted when a rate limit is set.
+     * @param key The identifier for the rate limit.
+     * @param maxAmount The maximum allowed amount for the rate limit.
+     * @param slope The slope value used in the rate limit calculation.
+     * @param lastAmount The amount left available at the last update.
+     * @param lastUpdated The timestamp when the rate limit was last updated.
+     */
+    event RateLimitSet(
+        bytes32 indexed key,
+        uint256 maxAmount,
+        uint256 slope,
+        uint256 lastAmount,
+        uint256 lastUpdated
+    );
+
+    /**
+     * @dev Emitted when a rate limit is triggered.
+     * @param key The identifier for the rate limit.
+     * @param amountToDecrease The amount to decrease from the current rate limit.
+     * @param oldLimit The previous rate limit value before triggering.
+     * @param newLimit The new rate limit value after triggering.
+     */
+    event RateLimitDecreaseTriggered(
+        bytes32 indexed key,
+        uint256 amountToDecrease,
+        uint256 oldLimit,
```

Comment:
Should be consistent across the whole codebase with this (e.g., `currentRateLimit` and `newLimit` in `triggerRateLimitDecrease`), lets always use `rateLimit`, so here `oldLimit => oldRateLimit, newLimit => newRateLimit`

Reply:
fixed